### PR TITLE
Add sac check to blob upsert

### PR DESCRIPTION
**Pull request by c-du**
## Description

Add sac check to upsert and delete.

These two operations require both read/write permissions. If a caller holds only one permission, it will lead to inconsistency.
1. If a caller has write permission but no read permission, upsert will lead to leaking large object.
2. If a caller has read permission but no write permission, delete will lead to a blob without large object.
This PR fixes that by augment ctx with extra read permission.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: central/blob/datastore/store/store.go [hunk 1]
```diff
@@ -46,6 +51,15 @@ func wrapRollback(ctx context.Context, tx *pgPkg.Tx, err error) error {
 
 // Upsert adds a blob to the database
 func (s *storeImpl) Upsert(ctx context.Context, obj *storage.Blob, reader io.Reader) error {
+	if err := sac.VerifyAuthzOK(scopeChecker.WriteAllowed(ctx)); err != nil {
+		return err
+	}
+	// Augment permission because we require read permission internally
+	ctx = sac.WithGlobalAccessScopeChecker(ctx,
```
Comment (dashrews78) [inline central/blob/datastore/store/store.go:58]: I'm not sure why we need the augmentation.  We have to have write to upsert and as far as I know, our SAC choices are `Access_NO_ACCESS`, `Access_READ_ACCESS`, and `Access_READ_WRITE_ACCESS`.  Only `Access_READ_WRITE_ACCESS` gives us write and it also has read.  If it doesn't we may have other issues with SAC.  Unless we need this augmentation for the scope key.  And then I wonder if we are allowing more than we should.

Comment (c-du) [reply central/blob/datastore/store/store.go:58]: That was what I thought, but you may run the test by removing the product side of changes and keep the test side of changes and run it. Some comments also pointing that. Access_READ_WRITE_ACCESS means write permission.

Comment (dashrews78) [reply central/blob/datastore/store/store.go:58]: In that case, it might be worth adding the SAC team to review.

Comment (c-du) [reply central/blob/datastore/store/store.go:58]: Right, @rhybrillou It looks like Access_READ_WRITE_ACCESS is not intuitive to its name. Do you know any background of it?

Comment (connorgorman) [inline central/blob/datastore/store/store.go:58]: Let's just do sac.WithAllAccess(ctx) for both these cases, but this is safe so 👍

Comment (c-du) [reply central/blob/datastore/store/store.go:58]: This is not needed to make it all access, so keep it as is?

**Review by connorgorman — APPROVED** [review]

**Pull request merged**